packages/@aws-cdk/aws-codebuild/lib/project.ts

```diff
@@ -588,6 +624,7 @@ export class Project extends ProjectBase {
   private readonly buildImage: IBuildImage;
   private readonly _secondarySources: BuildSource[];
   private readonly _secondaryArtifacts: BuildArtifacts[];
+  private securityGroups: ISecurityGroup[] = [];
```

Comment:
This should probably be called `_securityGroups`, and you should have a `public get securityGroups(): ISecurityGroup[]` property that returns a defensive copy of this array. Otherwise, the customer will not have access to the SecurityGroup you create implicitly, and so never have a chance to add egress rules if `allowAllOutbound` is false, kind of defeating the purpose of that property.